lib/omniauth/strategies/slack.rb

```diff
@@ -34,16 +33,25 @@ class Slack < OmniAuth::Strategies::OAuth2
           team: raw_info['team'],
           user: raw_info['user'],
           team_id: raw_info['team_id'],
-          team_domain: team_info['team']['domain'],
+          team_domain: team_info['team'].fetch('domain'),
           user_id: raw_info['user_id'],
-          is_admin: user_info['user']['is_admin'],
-          is_owner: user_info['user']['is_owner'],
-          time_zone: user_info['user']['tz']
+          is_admin: user_info['user'].fetch('is_admin'),
+          is_owner: user_info['user'].fetch('is_owner'),
+          time_zone: user_info['user'].fetch('tz')
```

Comment:
It's a style thing, with `fetch` you will get and error showing you which key was `nil` otherwise you would get `#<NoMethodError: undefined method `[]' for nil:NilClass>` which is not useful for these errors with hash keys.
I saw the bug report but to properly fix it we should remove all the team and user info from the info hash and leave it just in the extra, but that will be a backward incompatible change and I don't know what you think about it.
Meanwhile I could make the changes you did by invoking `to_h`